main.py

```diff
@@ -1,38 +1,58 @@
 import os
-import pprint
-from argparse import Namespace
 
-import yaml
+from omegaconf import DictConfig, OmegaConf
+
+
+def load_embedded_configs(config_path: str, cli_args: DictConfig) -> DictConfig:
+    """Loads all 'extends' embedded configs and merge them with the cli overwrites."""
+
+    cfgs = [OmegaConf.load(config_path)]
+    while "extends" in cfgs[0]:
+        extends_path = cfgs[0]["extends"]
+        del cfgs[0]["extends"]
+
+        # go to parents configs until defaults.py is reached
+        if extends_path != "lagrangebench/defaults.py":
+            cfgs = [OmegaConf.load(extends_path)] + cfgs
+        else:
+            from lagrangebench.defaults import defaults
+
+            cfgs = [defaults] + cfgs
+            break
```

Comment:
Cool. Not intuitive at first since `defaults.py` is the only non-yaml file you can specify as extend. Do you insist in having these nested extends? This could be as simple as always merging the defaults to the passed config

Reply:
I do insist on the option of having nested configs, and I'm still not too happy about not having the base.yaml for every dataset. 

Do you think that always having defaults.py as the lowest priority overwrite is a good idea? Then if we use one level of configs, there would be no example of how to even use "extends: ".

Reply:
ok, I'll use a special string to denote the lagrangebench default configs